tests/test_checker.py

```diff
@@ -437,11 +439,19 @@ async def test_check(self):
             relative_redirect.new_version.url,
             "https://httpbingo.org/base64/MzAtNTAgZmVyYWwgaG9ncyEK",
         )
-        self.assertEqual(
+        # XXX: Gzipped response will have size/checksum different from plain; match any
```

Comment:
Isn't the intention of the patch to make these deterministically have the checksum of the compressed content? In which case, why does the test allow both?

Reply:
We can't tell httpbingo to encode (or not) the content. Currently httbingo feeds us gzipped response, but I won't be surprised if it suddenly starts to feed plain, thus here check for both.

Reply:
Can we put this in the comment?

Reply:
Sure, done.

Reply:
If you know where we can get a response with well-known body that is deterministically gzipped, we can add it to tests.

Reply:
Probably the sustainable fix here is to put an HTTP server into the test suite and serve data locally. e.g. https://pytest-httpserver.readthedocs.io/en/latest/